src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java

```diff
@@ -308,4 +308,9 @@ protected void onPause() {
 
         BusProvider.getBus().unregister(this);
     }
+
+    @Override
+    public void onBackPressed() {
+        finishAffinity();
```

Comment:
I believe this is unnecessary.

Reply:
There are 2 cases : 
1. When the user is not logged in and its also the first time launch - In this case, we will have Introduction Activity and Navigation Activity in the back stack. On pressing back button first Introduction Activity will be popped then on second press Navigation Activity (with a blank screen).
2. When the user is not logged in and not a first time launch - In this, we would have Navigation Activity in back stack.

So this finishes all activities in stack and works well both of these cases

Reply:
Ok I understand the use cases. This seems fine.